ds3/ds3.py

```diff
@@ -6777,15 +6767,11 @@ class GetObjectResponse(AbstractResponse):
   
   def process_response(self, response):
     self.__check_status_codes__([200, 206])
-    localFile = None
-    if self.request.stream:
-      localFile = self.request.stream
-    else:
-      localFile = open(self.request.effective_file_name, "wb")
+    stream = self.request.stream
     if self.request.offset:
-      localFile.seek(self.request.offset, 0)
-    localFile.write(response.read())
-    localFile.close()
+      stream.seek(self.request.offset, 0)
```

Comment:
The response object shouldn't do any seeking on the stream.  The stream passed in needs to already be at the correct offset.